packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -182,6 +182,14 @@ export function updateWrapper(element: Element, props: Object) {
     } else if (node.value !== '' + value) {
       node.value = '' + value;
     }
+  } else if (
+    (props.type === 'submit' || props.type === 'reset') &&
+    (value === null || value === undefined)
+  ) {
+    // Submit/reset inputs need the attribute removed completely to avoid
+    // blank-text buttons.
+    node.removeAttribute('value');
```

Comment:
If I remove this line none of the tests seem to fail. Please make sure the tests cover all lines.